caiman/source_extraction/cnmf/lazy_arrays.py

```diff
@@ -0,0 +1,439 @@
+from warnings import warn
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import *
+from itertools import product as iter_product
+from time import time
+
+
+import numpy as np
+from scipy.sparse import csc_matrix
+import h5py
+
+
+# TODO: Is there a better way to do this?
+slice_or_int_or_range = Union[int, slice, range]
```

Comment:
Wondering what the convention is; do people normally use titlecase for types like this?